### ISSUE TITLE
Add custom Evaluator

### ISSUE THREAD

**Issue by adamflott** [h2non/gentleman-retry#4]
Currently, a fixed Evaluator can be used, it would be nice to be able to provide our own.

**Comment by h2non** [comment]
You can do it. Just implement the following interface and pass it to the `New` constructor:
https://github.com/h2non/gentleman-retry/blob/master/retry.go#L36-L39

**Comment by adamflott** [comment]
No, I meant, I would like to use a custom https://github.com/h2non/gentleman-retry/blob/master/retry.go#L50

**Comment by h2non** [comment]
You can overwrite the exported variable from `gentleman-retry` package:
```go
retry.Evaluator = func(err error, res *http.Response, req *http.Request) error {
  // ...
}
```

If you look for a client-specific evaluator, that's not currently allowed, but it can be easily done by exporting a retrier struct field so you can overwrite it, or also done via custom transport. PR are welcome.
